### ISSUE TITLE
Weird crash when using Distributions in parallel

### ISSUE THREAD

**Issue by tlnagy** [JuliaStats/Distributions.jl#492]
```{.julia}
julia> addprocs(8)
8-element Array{Int64,1}:
 2
 3
 4
 5
 6
 7
 8
 9

julia> @everywhere using Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions
WARNING: replacing module Distributions

julia> @everywhere println(Distributions.Sampleable{Univariate, Discrete})
Distributions.Sampleable{Distributions.Univariate,Distributions.Discrete}
	From worker 4:	Distributions.Sampleable{Distributions.Univariate,Distributions.Discrete}
	From worker 3:	Distributions.Sampleable{Distributions.Univariate,Distributions.Discrete}
	From worker 2:	Distributions.Sampleable{Distributions.Univariate,Distributions.Discrete}
ERROR: On worker 5:
TypeError: Sampleable: in F, expected F<:Distributions.VariateForm, got Type{Distributions.Univariate}
 in eval at /usr/local/Cellar/julia/0.4.5/lib/julia/sys.dylib
 in anonymous at multi.jl:1394
 in anonymous at multi.jl:923
 in run_work_thunk at multi.jl:661
 [inlined code] from multi.jl:923
 in anonymous at task.jl:63
 in remotecall_fetch at multi.jl:747
 in remotecall_fetch at multi.jl:750
 in anonymous at multi.jl:1396

...and 4 other exceptions.

 in sync_end at /usr/local/Cellar/julia/0.4.5/lib/julia/sys.dylib
 in anonymous at multi.jl:1405

julia> versioninfo()
Julia Version 0.4.5
Commit 2ac304d* (2016-03-18 00:58 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin15.4.0)
  CPU: Intel(R) Core(TM) i7-4850HQ CPU @ 2.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```
What's happening here?

**Comment by tlnagy** [comment]
This is using the latest stable version of Distributions.jl

**Comment by andreasnoack** [comment]
Does it also happen if you just do `using Distributions` instead of `@everywhere using Distributions`?

**Comment by tlnagy** [comment]
Then Distributions is out of scope?

```{.julia}
julia> addprocs(8)
8-element Array{Int64,1}:
 2
 3
 4
 5
 6
 7
 8
 9

julia> using Distributions

julia> @everywhere println(Distributions.Sampleable{Univariate, Discrete})
Distributions.Sampleable{Distributions.Univariate,Distributions.Discrete}
ERROR: On worker 2:
UndefVarError: Univariate not defined
 in eval at /usr/local/Cellar/julia/0.4.5/lib/julia/sys.dylib
 in anonymous at multi.jl:1394
 in anonymous at multi.jl:923
 in run_work_thunk at multi.jl:661
 [inlined code] from multi.jl:923
 in anonymous at task.jl:63
 in remotecall_fetch at multi.jl:747
 in remotecall_fetch at multi.jl:750
 in anonymous at multi.jl:1396

...and 7 other exceptions.

 in sync_end at /usr/local/Cellar/julia/0.4.5/lib/julia/sys.dylib
 in anonymous at multi.jl:1405
```

**Comment by tlnagy** [comment]
Maybe I'm misunderstanding the docs, but I was under the impression that `using Distributions` only brings the module into scope on the first worker. I thought that adding `@everywhere` would force it to be in scope on all workers. Is there a different way of doing that?

**Comment by andreasnoack** [comment]
Ah. You'll have to do something like
```jl
using Distributions
@everywhere using Distributions
```
I'm not sure why.

**Comment by tlnagy** [comment]
💯 thank you, that does the trick. I don't think that docs make that clear enough.
